### PR TITLE
Increase PFCP buffer size and update `go-pfcp`

### PR DESCRIPTION
**Pull request by pudelkoM**
We observed a pfcp agent crash when PFCP messages are larger than a certain size. The UDP receive buffer is too small and causes the message to be truncated.
Then, the library code reads beyond the buffer and causes the panic seen below:

```
panic: runtime error: slice bounds out of range [:50] with capacity 41
goroutine 13 [running]:
github.com/wmnsk/go-pfcp/ie.(*IE).UnmarshalBinary(0x300c00003fa00, {0xc0002eebb3, 0x2, 0x3})
    /pfcpiface/vendor/github.com/wmnsk/go-pfcp/ie/ie.go:367 +0x187
github.com/wmnsk/go-pfcp/ie.Parse({0xc0002eebb3, 0x29, 0x29}) 
    /pfcpiface/vendor/github.com/wmnsk/go-pfcp/ie/ie.go:339 +0x48
github.com/wmnsk/go-pfcp/ie.ParseMultiIEs({0xc0002ee610, 0x5dc, 0x5dc}) 
    /pfcpiface/vendor/github.com/wmnsk/go-pfcp/ie/ie.go:628 +0x8c 
github.com/wmnsk/go-pfcp/message.(*SessionEstablishmentRequest).UnmarshalBinary(0xc000232000, {0xc0002ee600, 0x51a530, 0xc000298008}) 
    /pfcpiface/vendor/github.com/wmnsk/go-pfcp/message/session-establishment-request.go:291 +0x6e
github.com/wmnsk/go-pfcp/message.Parse({0xc0002ee600, 0x5dc, 0x5dc}) 
    /pfcpiface/vendor/github.com/wmnsk/go-pfcp/message/message.go:116 +0x3ab
main.pfcpifaceMainLoop(0xc00037e500, {0xc0000ce240, 0x0}, {0xc0000ce250, 0xf}, {0xc0000ce260, 0x7}, {0x0, 0x0})
    /pfcpiface/conn.go:133 +0x8e5
created by main.main
    /pfcpiface/main.go:231 +0x94f
```

The proposed solution is to increase the buffer size to `4096` ~and check the PFCP header length field~. The length check for IE parsing needs to happen in `pfcp-go`.

image

### CODE REVIEW

File: pfcpiface/conn.go [hunk 1]
```diff
@@ -148,8 +148,20 @@ func (pConn *PFCPConn) Serve() {
...
+			// Check that the PFCP message did not get truncated.
+			header, err := message.ParseHeader(buf)
+			if err != nil {
+				log.Errorln("could not parse PFCP message header: ", err)
+				continue
+			}
+			if header.Length > uint16(n) {
+				log.Errorln("receive buffer too small for PFCP message")
+				continue
+			}
```
Comment (pudelkoM) [inline pfcpiface/conn.go:162]: This check is already happening in the library: https://github.com/omec-project/upf-epc/blob/a6b6a4e29cc56c902cf7ca03b10a7d01f9cd98ae/pfcpiface/vendor/github.com/wmnsk/go-pfcp/message/session-establishment-request.go#L281-L286

**Comment by pudelkoM** [comment]
Managed to reproduce the crash in a small test, see latest commit:

```
=== RUN   Test
--- FAIL: Test (0.00s)
panic: runtime error: slice bounds out of range [:20] with capacity 5 [recovered]
	panic: runtime error: slice bounds out of range [:20] with capacity 5

goroutine 34 [running]:
testing.tRunner.func1.2({0x15f2ae0, 0xc0003dc018})
	/usr/local/Cellar/go/1.17.2/libexec/src/testing/testing.go:1209 +0x24e
testing.tRunner.func1()
	/usr/local/Cellar/go/1.17.2/libexec/src/testing/testing.go:1212 +0x218
panic({0x15f2ae0, 0xc0003dc018})
	/usr/local/Cellar/go/1.17.2/libexec/src/runtime/panic.go:1038 +0x215
github.com/wmnsk/go-pfcp/ie.(*IE).UnmarshalBinary(0xc00006c6c0, {0xc0002a8d88, 0x5, 0x153f8e0})
	/Users/max/go/pkg/mod/github.com/wmnsk/go-pfcp@v0.0.11/ie/ie.go:367 +0x187
github.com/wmnsk/go-pfcp/ie.Parse({0xc0002a8d88, 0x5, 0x5})
	/Users/max/go/pkg/mod/github.com/wmnsk/go-pfcp@v0.0.11/ie/ie.go:339 +0x48
github.com/omec-project/upf-epc/pfcpiface.Test(0xc000292680)
	/Users/max/projects/upf-epc/pfcpiface/conn_test.go:12 +0x4d
testing.tRunner(0xc000292680, 0x1680060)
	/usr/local/Cellar/go/1.17.2/libexec/src/testing/testing.go:1259 +0x102
created by testing.(*T).Run
	/usr/local/Cellar/go/1.17.2/libexec/src/testing/testing.go:1306 +0x35a


Process finished with the exit code 1
```

File: pfcpiface/conn_test.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+	parsedIe, err := ie.Parse(buf)
+	if err != nil {
+		log.Error(err)
+		t.FailNow()
+	}
+	log.Warnln("Type:", parsedIe.Type)
+	log.Warnln("Length:", parsedIe.Length)
+	log.Warnln(parsedIe)
+}
```
Comment (pudelkoM) [inline pfcpiface/conn_test.go:20]: Delete before merging

**Review by krsna1729 — APPROVED** [review]

**Review by krsna1729 — DISMISSED** [review]

**Comment by pudelkoM** [comment]
retest this

**Comment by pudelkoM** [comment]
retest this please

**Review by krsna1729 — APPROVED** [review]

**Pull request merged**